src/AWS.Deploy.CLI/Commands/ServerModeCommand.cs

```diff
@@ -136,7 +136,9 @@ private async Task<bool> WaitOnParentPid(CancellationToken token)
             {
                 try
                 {
-                    Process.GetProcessById((int)_parentPid);
+                    var process = Process.GetProcessById((int)_parentPid);
+                    if (process.HasExited)
+                        return true;
                     await Task.Delay(1000, token);
                 }
                 catch
```

Comment:
Can you remove the infinite while and use `WaitForExit`?

https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexit?view=net-5.0#system-diagnostics-process-waitforexit(system-int32)

Reply:
WaitForExit is a blocking call. I reworked it to use events to remove the while loop.